homeassistant/components/hunterdouglas_powerview/shade_data.py

```diff
@@ -76,3 +73,11 @@ def store_group_data(self, shade_data: PowerviewData) -> None:
     def update_shade_position(self, shade_id: int, new_position: ShadePosition) -> None:
         """Update a single shades position."""
         copy_position_data(new_position, self.get_shade_position(shade_id))
+
+    def update_shade_velocity(self, shade_id: int, shade_data: ShadePosition) -> None:
+        """Update a single shades velocity."""
```

Comment:
The old code had this also
```suggestion
        """Update a single shades velocity."""
        if shade_id not in self.positions:
            self.positions[shade_id] = ShadePosition()
            
```

Reply:
I've just learned none of my reviews ever made it through before 😄 

Didnt know i needed to click the "Finish Review" in the top right

Reply:
Calling `get_shade_position` should already do that

https://github.com/home-assistant/core/blob/ae0b41f7a79c04c8ddcab3fa9cf26f0acfbe7889/homeassistant/components/hunterdouglas_powerview/shade_data.py#L50

Reply:
It does - but velocity is stripped out now

I believe the line just prevents an error in the event a entity doesn't exist in that object yet

``get_shade_position`` should ensure the element exists in the co-ordinator as long as that is called before the number entity is created

You would understand more about that architecture 

Essentially the number element would not have the ability to create the entry in the coordinator

Reply:
> I've just learned none of my reviews ever made it through before 😄
> 
> Didnt know i needed to click the "Finish Review" in the top right

No worries, thanks for taking a look

Reply:
> Essentially the number element would not have the ability to create the entry in the coordinator

`__init__.py` will always load the data into the coordinator before it forwards setup to the `number`, `sensor`, etc platforms

So if I follow it correctly (hopefully) when this gets called

https://github.com/home-assistant/core/pull/114337/files#diff-568965c8d5d5cf43cccbe9c150cb9b3507219cf048043921147a9ee67181cc72R44

https://github.com/home-assistant/core/pull/114337/files#diff-24d5bd6c997d13b855af682446cd21d27db03e941601c9bc42abbfcd7b179983R83

The data that was set in
https://github.com/home-assistant/core/blob/b1fa6f50962c305d50345f7be5c8a3a026ad8d0b/homeassistant/components/hunterdouglas_powerview/__init__.py#L101

get pulled out
https://github.com/home-assistant/core/blob/b1fa6f50962c305d50345f7be5c8a3a026ad8d0b/homeassistant/components/hunterdouglas_powerview/shade_data.py#L52

and than the velocity will now be added to it

https://github.com/home-assistant/core/blob/b1fa6f50962c305d50345f7be5c8a3a026ad8d0b/homeassistant/components/hunterdouglas_powerview/shade_data.py#L83